src/modules/Builder/components/Configuration/Configuration.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+
+import { wirelessSupportOS, wirelessSupportDevices } from "./config";
+
+import Main from "./components/Main/Main";
+import Wireless from "./components/Wireless/Wireless";
+
+export default class Configuration extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            buttonStates: {
+                mainHostName: true,
+                wirelessPSK: true,
+            },
+        };
+
+        this.serverRefs = {
+            main: React.createRef(),
+        };
+
+        this.buttonStateCallback = this.buttonStateCallback.bind(this);
+    }
+
+    executeState() {
+        const { builderCallback } = this.props;
+        builderCallback();
+        Object.keys(this.serverRefs).forEach((key) => {
+            this.serverRefs[key].current.syncConfigParams();
+        });
+    }
+
+    buttonStateCallback(name, buttonVariable) {
+        const { builderCallback } = this.props;
+        const { buttonStates } = this.state;
+        buttonStates[name] = buttonVariable;
+        if (Object.values(buttonStates).every((elem) => elem === true)) {
+            builderCallback({
+                state: "buttonState",
+                buttonState: false,
+            });
+        } else {
+            builderCallback({
+                state: "buttonState",
+                buttonState: true,
+            });
+        }
+    }
+
+    checkWirelessSupport() {
+        const { os, device } = this.props;
+        if (wirelessSupportOS.includes(os) && wirelessSupportDevices.includes(device)) {
+            this.serverRefs.wireless = React.createRef();
+            return true;
+        }
+        return false;
+    }
+
+    render() {
+        return (
+            <div className="builder-configuration-stage" key="cards">
+                <Main
+                    buttonStateCallback={this.buttonStateCallback}
+                    ref={this.serverRefs.main}
+                />
+                {(this.checkWirelessSupport())
+                    && (
+                        (
+                            <Wireless
+                                buttonStateCallback={this.buttonStateCallback}
+                                ref={this.serverRefs.wireless}
+                            />
+                        )
+                    )}
+            </div>
+        );
+    }
+}
+
+Configuration.propTypes = {
```

Comment:
Add these props to all components of the builder stages.

Reply:
PackageList stage doesnt use these props, hence no verification is needed, eslint points on this